pkg/users/password.jsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import cockpit from 'cockpit';
+import React from 'react';
+import { FormGroup, Progress, ProgressSize, ProgressMeasureLocation, TextInput } from '@patternfly/react-core';
+
+const _ = cockpit.gettext;
+
+export function password_quality(password, force) {
+    return new Promise((resolve, reject) => {
+        cockpit.spawn('/usr/bin/pwscore', { err: "message" })
+                .input(password)
+                .done(function(content) {
+                    var quality = parseInt(content, 10);
+                    if (quality === 0) {
+                        reject(new Error(_("Password is too weak")));
+                    } else if (quality <= 33) {
+                        resolve("weak");
+                    } else if (quality <= 66) {
+                        resolve("okay");
+                    } else if (quality <= 99) {
+                        resolve("good");
+                    } else {
+                        resolve("excellent");
+                    }
```

Comment:
I feel like this groups has lost meaning. We should just reject on 0 and otherwise `resolve(quality)`. Then we would just do that check what type of progress bar we want but we would use the number from `pwscore` and not our made up ranges. It could work more linearly as you type. (It used to be here as these words were also css classes).

Reply:
Done.